programs/client/TestHint.cpp

```diff
@@ -0,0 +1,105 @@
+#include "TestHint.h"
+
+#include <Common/Exception.h>
+#include <Common/ErrorCodes.h>
+#include <IO/ReadBufferFromString.h>
+#include <IO/ReadHelpers.h>
+#include <Parsers/Lexer.h>
+
+namespace
+{
+
+/// Parse error as number or as a string (name of the error code const)
+int parseErrorCode(DB::ReadBufferFromString & in)
+{
+    int code = -1;
+    String code_name;
+
+    auto * pos = in.position();
+    tryReadText(code, in);
+    if (pos != in.position())
+    {
+        return code;
```

Comment:
We can also check that code is non-negative number and less that ErrorCodes::end(), but it's not cruisual